lib/sqlite-store.js

```diff
@@ -273,26 +366,28 @@ module.exports = function (options) {
 
     stm.text = 'INSERT INTO ' + escapeStr(table) + ' (' + escapeStr(fields) + ') values (' + escapeStr(params) + ')'
     stm.values = values
+
     return stm
   }
 
 
-  var updatestm = function (ent) {
+  var updatebyidstm = function (ent, schema) {
     var stm = {}
     var table = tablename(ent)
-    var fields = ent.fields$()
     var entp = makeentp(ent)
 
     var values = {}
     var params = []
 
-    fields.forEach(function (field) {
-      if (!(_.isUndefined(ent[field]) || _.isNull(ent[field]))) {
-        var fieldPlaceholder = '$' + field
-        values[fieldPlaceholder] = entp[field]
-        params.push(field + ' = ' + fieldPlaceholder)
-      }
-    })
+    for (const col in schema) {
+      const fieldPlaceholder = '$' + col
+
+      values[fieldPlaceholder] = col in entp
+        ? entp[col]
+        : null
```

Comment:
SQL requires all columns to be updated to NULL explicitly, unlike in NoSQL databases.